cmd/cloud/dbt.go

```diff
@@ -0,0 +1,150 @@
+package cloud
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	cloud "github.com/astronomer/astro-cli/cloud/deploy"
+	"github.com/astronomer/astro-cli/cloud/deployment"
+	"github.com/astronomer/astro-cli/config"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	mountPath      string
+	dbtProjectPath string
+
+	DeployBundle = cloud.DeployBundle
+)
+
+const (
+	dbtDeployExample = `
+Specify the ID of the Deployment on Astronomer you would like to deploy this dbt project to:
+
+  $ astro dbt deploy -d <deployment ID>
+
+Menu will be presented if you do not specify a deployment ID:
+
+  $ astro dbt deploy
+`
+
+	dbtDefaultMountPathPrefix = "/usr/local/airflow/dbt/"
+	dbtProjectYmlFilename     = "dbt_project.yml"
+	dbtBundleType             = "dbt"
+)
+
+func newDbtCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "dbt",
+		Short: "Manage your dbt projects deployed to Deployments running on Astronomer",
+	}
+	cmd.AddCommand(
+		newDbtDeployCmd(),
+	)
+	return cmd
+}
+
+func newDbtDeployCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "deploy",
+		Short:   "Deploy your dbt project to a Deployment on Astro",
+		Long:    "Deploy your dbt project to a Deployment on Astro. This command bundles your dbt project files and uploads it to your Deployment.",
+		Args:    cobra.NoArgs,
+		RunE:    deployDbt,
+		Example: dbtDeployExample,
+	}
+
+	cmd.Flags().StringVarP(&mountPath, "mount-path", "m", "", "Path to mount dbt project in Airflow, for reference by DAGs. Default /usr/local/dbt/{dbt project name}")
+	cmd.Flags().StringVarP(&dbtProjectPath, "project-path", "p", "", "Path to the dbt project to deploy. Default current directory")
+	cmd.Flags().StringVar(&workspaceID, "workspace-id", "", "Workspace for your Deployment")
+	cmd.Flags().StringVarP(&deploymentID, "deployment-id", "d", "", "ID of the Deployment to deploy to")
+	cmd.Flags().StringVarP(&deploymentName, "deployment-name", "n", "", "Name of the Deployment to deploy to")
+	cmd.Flags().StringVarP(&deployDescription, "description", "", "", "Description to store on the deploy")
+	cmd.Flags().BoolVarP(&waitForDeploy, "wait", "w", false, "Wait for the Deployment to become healthy before ending the command")
+
+	return cmd
+}
+
+func deployDbt(cmd *cobra.Command, args []string) error {
+	cmd.SilenceUsage = true
+
+	// if the dbt project path is not provided, use the current directory
+	if dbtProjectPath == "" {
+		dbtProjectPath = config.WorkingPath
+	}
+
+	// check that there is a valid dbt project at the dbt project path
+	dbtProjectYamlPath := filepath.Join(dbtProjectPath, dbtProjectYmlFilename)
+	if _, err := os.Stat(dbtProjectYamlPath); os.IsNotExist(err) {
```

Comment:
Can we do this as part of PreRunE and move this check to utils

Reply:
What is the advantage of calling it from `PreRunE`? I like having these checks in the `Run` function so it's in one place. `astro deploy` has a few checks in the `Run` function and that doesn't seem to be an issue. I will move the logic to a util though.

Reply:
I know, but we don't have to use it because it exists. In my view it isn't helpful to have an artificial split like that. We don't use it for most checks in `astro deploy` -- it only checks that it is running in a project directory, not whether the workspace can be determined, or that the `dags` and `image` flags aren't used together, or that the config can be saved, or that there aren't uncommitted changes.

Reply:
I just checked that the path is an user input and not derived from where this command is executed. This flow is different from how dag/image deploys work. We might want to confirm this with product once just so we are on the same page. This was not how i was imagining this check would work

My comment was running the preRunE check to confirm the current directory path is a dbt project or not.

Reply:
> the path is an user input and not derived from where this command is executed

Not quite, the code directly above this uses the current working directory by default, but the user can override it with a flag. This seems pretty useful as an option for CI where you've cloned a repo and want to deploy a dbt project in some subdirectory of it.

Reply:
Afaik we also do this in `astro deploy` with the hidden (?) `dags-path` flag.

Reply:
gotcha! moving this to utils should be good then

Reply:
I've made it a separate function in the file so we can reuse it for `astro dbt delete`.